### ISSUE TITLE
Question: transfer leader prevents fix down peer

### ISSUE THREAD

**Issue by Ryan-Git** [tikv/pd#3226]
As per my understanding, peer down seconds reported by TiKV is reseted upon leader change. 

If leader transfers back and forth between the two healthy peers within 30min(which is the default max store down time), PD would have no change to fix this down peer, am I right?

**Comment by rleungx** [comment]
Yes, it seems that we indeed have this problem. Can you confirm it? @BusyJay

**Comment by BusyJay** [comment]
Can PD fix them by removing down store?

**Comment by Ryan-Git** [comment]
manually offline the store of course fix this, but it's better to be automatic.

**Comment by BusyJay** [comment]
Down peers shows because of down stores, right? So you don't have to mark it offline manually.

**Comment by Ryan-Git** [comment]
as per 3.0.5 replica checker(actually for the latest master it seems similar), I can neither find logic which replace peers in down stores(in `checkDownPeer`), nor store score penalty for down stores(in `checkBestReplacement`), just `checkOfflinePeer`. Am I miss anything?

I previously thought it was because PD should always respect information reported, to prevent wrong decision due to asymmetric network partition.

**Comment by rleungx** [comment]
> checkDownPeer loops through down peers and finally check peer's down seconds. checkBestReplacement excludes down stores by health filter. So in this scenario, it seems only checkOfflinePeer could work, which is manually marked.

You're right. For now, in such a case, we need to offline this store manually. I think we can fix this situation by not depending on the down peers. Instead, we could check if the store of these regions is down. WDYT @Ryan-Git

**Comment by Ryan-Git** [comment]
> > checkDownPeer loops through down peers and finally check peer's down seconds. checkBestReplacement excludes down stores by health filter. So in this scenario, it seems only checkOfflinePeer could work, which is manually marked.
> 
> You're right. For now, in such a case, we need to offline this store manually. I think we can fix this situation by not depending on the down peers. Instead, we could check if the store of these regions is down. WDYT @Ryan-Git

I think it's fine. Even with asymmetric network partition, we're still safe. 

If so, what's the purpose of down peers originally?

**Comment by rleungx** [comment]
Do you have any idea? @nolouch @disksing

**Comment by disksing** [comment]
It was intended to handle asymmetric network partition and avoid redundant actions.

**Comment by Ryan-Git** [comment]
> It was intended to handle asymmetric network partition and avoid redundant actions.

So which do you think is better, depends on store down time or down peer seconds?

**Comment by disksing** [comment]
I think the current behaviour is reasonable, which is only move peer after both down time time out.

**Comment by Ryan-Git** [comment]
> I think the current behavior is reasonable, which is only move peer after both down time time out.

This decision seems to trade performance with availability?

I thought long time asymmetric network partition should be quite rare so personally prefer redundant actions. Not quite a problem in prod any way. But we randomly request transfer leader in test env, which prevents pd from remove such peer. Any idea for such scenario?

**Comment by disksing** [comment]
em, that's a good point. I think we may introduce a new option to control the behaviour.
